### PR TITLE
feat:Added support for django3.2

### PR DESCRIPTION
**Pull request by Jawayria**
JIRA: https://openedx.atlassian.net/browse/BOM-2618

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,55 +1,56 @@
 [tox]
-envlist = py38-django{22,30,31}-celery{44,50}-drf{39,310,311,312},quality,docs
+envlist = py38-django{22,30,31,32}-celery{44,50}-drf{311,312},quality,docs
```
Comment (awais786) [inline tox.ini:2]: In platform `djangorestframework==3.12.4` version is on prod, we can safely drop earlier version and add new one.
Try some thing like this `-drf{312,latest}`

File: tox.ini [hunk 2]
```diff
@@ -1,55 +1,56 @@
...
+    django22: Django>=2.2,<2.3
+    django30: Django>=3.0,<3.1
+    django31: Django>=3.1,<3.2
+    django32: Django>=3.2,<3.3
+    drf39: djangorestframework>=3.9,<3.10
+    drf310: djangorestframework>=3.10,<3.11
+    drf311: djangorestframework>=3.11,<3.12
+    drf312: djangorestframework>=3.12,<3.13
+    celery44: -r{toxinidir}/requirements/celery44.txt
+    celery50: -r{toxinidir}/requirements/celery50.txt
```
Comment (awais786) [inline tox.ini:15]: ```drflatest: djangorestframework```

**Comment by Jawayria** [comment]
These warnings are raised during the execution of tests and are not fixed under the scope of this PR

```
1. RemovedInDjango40Warning: django.utils.translation.ugettext_lazy() is deprecated in favor of django.utils.translation.gettext_lazy().

2. RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
 
3. RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
     urlpatterns += [url(r'^admin/', admin.site.urls)]. 
```

**Comment by awais786** [comment]
> These warnings are raised during the execution of tests and were not fixed under the scope of this PR
> 
> ```
> 1. RemovedInDjango40Warning: django.utils.translation.ugettext_lazy() is deprecated in favor of django.utils.translation.gettext_lazy().
> 
> 2. RemovedInDjango40Warning: The providing_args argument is deprecated. As it is purely documentational, it has no replacement. If you rely on this argument as documentation, you can move the text to a code comment or docstring.
>  
> 3. RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
>      urlpatterns += [url(r'^admin/', admin.site.urls)]. 
> ```

@jmbowman django40 warnings are out of scope ?

**Review by awais786 — APPROVED** [review]

File: tox.ini [hunk 3]
```diff
@@ -1,55 +1,54 @@
...
-	make
-	rm
+    make
+    rm
 deps = 
-	-r{toxinidir}/requirements/doc.txt
+    -r{toxinidir}/requirements/doc.txt
 commands = 
-	doc8 --ignore-path docs/_build --ignore-path docs/rest_api.rst README.rst docs
-	rm -f docs/user_tasks.rst
```
Comment (mraarif) [inline tox.ini:27]: let's undo these indentation changes

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -2,25 +2,24 @@ name: Python CI
 
 on:
   push:
-    branches: [ master ]
+    branches: [master]
   pull_request:
     branches:
-      - '**'
```
Comment (mraarif) [inline .github/workflows/ci.yml:8]: same here

**Review by mraarif — APPROVED** [review]

**Comment by awais786** [comment]
One missing point in these PRs is running make upgrade with `django==3.2`
@jmbowman FYI

File: CHANGELOG.rst [hunk 5]
```diff
@@ -16,6 +16,14 @@ Unreleased
 
 *
 
+[2.1.0] - 2021-07-07
+~~~~~~~~~~~~~~~~~~~~
+
+Added
++++++
+
+* Support for django3.2
```
Comment (jmbowman) [inline CHANGELOG.rst:25]: And 3.0 and 3.1

File: setup.py [hunk 6]
```diff
@@ -80,6 +80,7 @@ def is_requirement(line):
         'Development Status :: 4 - Beta',
         'Framework :: Django',
         'Framework :: Django :: 2.2',
+        'Framework :: Django :: 3.2',
```
Comment (jmbowman) [inline setup.py:83]: Here also, if we're testing 3.0 and 3.1 we should probably claim to support them.

**Review by jmbowman — APPROVED** [review]
A couple of suggestions, otherwise 👍

**Pull request merged**